src/NuGetForUnity/Editor/NugetHelper.cs

```diff
@@ -709,6 +755,19 @@ internal static void UninstallAll()
             {
                 Uninstall(package);
             }
+            UnselectAll();
+        }
+
+        /// <summary>
+        ///     Uninstalls all of the currently installed and selected packages.
+        /// </summary>
+        internal static void UninstallAllSelected()
+        {
+            foreach (var package in selectedInstalledPackages.Values.ToList())
```

Comment:
You should be able to write this as a for loop that just calls Uninstall for each element and then after the loop you could just call UnselectAll().